src/libasr/pass/pass_utils.cpp

```diff
@@ -560,7 +560,25 @@ namespace LCompilers {
 
         ASR::expr_t* get_bound(ASR::expr_t* arr_expr, int dim, std::string bound,
                                 Allocator& al) {
+            ASR::ttype_t* x_mv_type = ASRUtils::expr_type(arr_expr);
+            ASR::dimension_t* m_dims;
+            int n_dims = ASRUtils::extract_dimensions_from_ttype(x_mv_type, m_dims);
+            bool is_data_only_array = ASRUtils::is_fixed_size_array(m_dims, n_dims) && ASRUtils::get_asr_owner(arr_expr) &&
+                                    ASR::is_a<ASR::StructType_t>(*ASRUtils::get_asr_owner(arr_expr));
```

Comment:
Does "data only array" in this context mean "FixedSizeArray" inside a struct?

Reply:
Yes.

Reply:
Ok. I would maybe document this in a comment, or maybe name the variable `fixed_array_inside_struct`, or something like that.